File: Code/Command_Communication/Command_csharp/Command_csharp/Command.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Command_csharp
{
    class Command
    {
        int id;
        string username;
        int typ;
        string filename;
        string parameter;
        byte[] data;

        public int Id
        {
            get
            {
                return id;
            }

            set
            {
                id = value;
            }
        }

        public string Username
        {
            get
            {
                return username;
            }

            set
            {
                username = value;
            }
        }

        public int Typ
        {
            get
            {
                return typ;
            }

            set
            {
                typ = value;
            }
        }

        public string Filename
        {
            get
            {
                return filename;
            }

            set
            {
                filename = value;
            }
        }

        public string Parameter
        {
            get
            {
                return parameter;
            }

            set
            {
                parameter = value;
            }
        }

        public byte[] Data
        {
            get
            {
                return data;
            }

            set
            {
                data = value;
            }
        }

        public string printInnerHeader()
        {
            return id + ";" + username + ";" + typ + ";" + filename + ";" + parameter + ";" + data;
        }

        public byte[] toByteArr()
        {
            string b64 = Convert.ToBase64String(Encoding.UTF8.GetBytes(printInnerHeader()));
            string header = "{" + b64 + "}";
            int len = header.Length + this.data.Length;
            byte[] rawCmd = new byte[len];
            for(int i=0;i< len;i++)
            {
                if (i < header.Length)
                {
                    rawCmd[i] = (byte)header[i];
                }
                else
                {
                    rawCmd[i] = (byte)this.data[i-header.Length];
                }
            }
            return rawCmd;
        }

        
    }
}

File: Code/Command_Communication/DataClient/src/main/Command.java
package main;

import java.util.Base64;

/**
 * Created by apolol92 on 02.01.2016.
 */
public class Command {
    public int Id;
    public String Username;
    public int Typ;
    public String Filename;
    public String Parameter;
    public byte[] Data;

    public String printInnerHeader()
    {
        return Id + ";" + Username + ";" + Typ + ";" + Filename + ";" + Parameter + ";" + Data;
    }
    public byte[] toByteArr()
    {
        String b64 = Base64.getEncoder().encodeToString(printInnerHeader().getBytes());
        String header = "{" + b64 + "}";
        int len = header.length() + this.Data.length;
        byte[] rawCmd = new byte[len];
        for(int i=0;i< len;i++)
        {
            if (i < header.length())
            {
                rawCmd[i] = header.getBytes()[i];
            }
            else
            {
                rawCmd[i] = (byte)this.Data[i-header.length()];
            }
        }
        return rawCmd;
    }
}
